src/Microsoft.Health.Dicom.Core/Features/Common/IFileStore.cs

```diff
@@ -70,4 +70,14 @@ public interface IFileStore
         VersionedInstanceIdentifier versionedInstanceIdentifier,
         FrameRange range,
         CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Asynchronously deletes a old DICOM file from the file store if the file exists.
+    /// </summary>
+    /// <param name="versionedInstanceIdentifier">The DICOM identifier.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>A task that represents the asynchronous delete operation.</returns>
+    Task DeleteOldFileIfExistsAsync(
+        VersionedInstanceIdentifier versionedInstanceIdentifier,
+        CancellationToken cancellationToken);
```

Comment:
nit: there should be a default value for the token

Reply:
Done